client/state/jetpack-connect/test/selectors.js

```diff
@@ -6,49 +6,257 @@ import { expect } from 'chai';
 /**
  * Internal dependencies
  */
-import { isCalypsoStartedConnection, getFlowType, getJetpackSiteByUrl, hasXmlrpcError } from '../selectors';
-
-const stateHasXmlrpcError = {
-	jetpackConnect: {
-		jetpackConnectAuthorize: {
-			authorizeError: {
-				message: 'transport error - HTTP status code was not 200 (502)'
-			},
-			authorizationCode: 'xxxx'
-		}
-	}
-};
-
-const stateHasNoError = {
-	jetpackConnect: {
-		jetpackConnectAuthorize: {
-			authorizeError: false
-		}
-	}
-};
-
-const stateHasNoAuthorizationCode = {
-	jetpackConnect: {
-		jetpackConnectAuthorize: {
-			authorizeError: {
-				message: 'Could not verify your request.'
-			}
-		}
-	}
-};
-
-const stateHasOtherError = {
-	jetpackConnect: {
-		jetpackConnectAuthorize: {
-			authorizeError: {
-				message: 'Jetpack: [already_connected] User already connected.'
-			},
-			authorizationCode: 'xxxx'
-		}
-	}
-};
+import {
+	getConnectingSite,
+	getAuthorize,
+	getAuthorizeQuery,
+	getAuthorizeSite,
+	getSessions,
+	getSSOSessions,
+	getSSO,
+	isCalypsoStartedConnection,
+	getFlowType,
+	getJetpackSiteByUrl,
+	hasXmlrpcError
+} from '../selectors';
 
 describe( 'selectors', () => {
+	describe( '#getConnectingSite()', () => {
+		it( 'should return undefined if user has not started connecting a site', () => {
+			const state = {
+				jetpackConnect: {}
+			};
+
+			expect( getConnectingSite( state ) ).to.be.undefined;
+		} );
+
+		it( 'should return the current connecting site if there is such', () => {
+			const jetpackConnectSite = {
+				url: '',
+				isFetching: true,
+				isFetched: false,
+				isRedirecting: false,
+				installConfirmedByUser: true,
+				isDismissed: false,
+				data: {
+					isDotCom: false,
+					notExists: false,
+					notWordPress: false,
+					notJetpack: false,
+					jetpackVersion: '4.3.1',
+					outdatedJetpack: false,
+					notActiveJetpack: false,
+					notConnectedJetpack: true,
+					alreadyOwned: true,
+					alreadyConnected: false
+				}
+			};
+			const state = {
+				jetpackConnect: {
+					jetpackConnectSite
+				}
+			};
+
+			expect( getConnectingSite( state ) ).to.eql( jetpackConnectSite );
```

Comment:
now I see this test, I think it would be a good idea if this should be a failing one. 

I'm not sure if we already cover this in our standardized data approach ( @aduth ? ), but I think selectors should return copies of the state subtrees, not a pointer to the subtree. That way the components couldn't modify anything in the state. 

That way, this test should be failing ... what the selector returns shouldn't be the same object that is stored on the state, just another one with the same content